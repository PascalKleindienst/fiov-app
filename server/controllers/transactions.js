// ========================================
// TRANSACTIONS CONTROLLER=================
// ========================================

// List all transactions of the current user.
module.exports.list_all_transactions = (req, res) => {
    req.app.locals.models.Transactions.findAll({ //findAndCountAll
        where: { user_id: req.user.id },
        include: ['category']
    })
        .then( (transactions) => {
            res.json(transactions);
        })
        .catch((err) => {
            res.send(err);
        });
};

// Create a new transaction for the current user.
module.exports.create_transaction = (req, res) => {
    const { description = '', note = '', amount = 0, createdAt = null } = req.body;

    req.app.locals.models.Transactions
        .create({ description, note, amount, createdAt, user_id: req.user.id })
        .then((transaction) => {
            res.json(transaction.get());
        })
        .catch((err) => {
            res.send(err);
        });
};
