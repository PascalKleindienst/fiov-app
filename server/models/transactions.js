// ========================================
// Transactions Model =====================
// ========================================
module.exports = (sequelize, DataTypes) => {
    const Transactions = sequelize.define('Transactions', {
        description: DataTypes.STRING,
        note: DataTypes.TEXT,
        amount: DataTypes.INTEGER
    }, {});

    Transactions.associate = function(models) {
        Transactions.belongsTo(models.Categories, {
            foreignKey: 'category_id',
            onDelete: 'CASCADE',
            as: 'category'
        });

        Transactions.belongsTo(models.User, {
            foreignKey: 'user_id',
            onDelete: 'CASCADE',
            as: 'user'
        });
    };
    
    return Transactions;
};