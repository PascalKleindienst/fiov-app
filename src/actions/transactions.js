// ==============================================
// TRANSACTIONS ACTIONS =========================
// ==============================================
import ajax from 'fetchival';

// =============== Add Transaction ===============

// =============== Remove Transaction ===============

// =============== Update Transaction ===============

// =============== Set Transactions ===============
export const SET_TRANSACTIONS = 'SET_TRANSACTIONS';

export const setTransactions = () => {
    return (dispatch) => {
        return ajax('/api/transactions', { mode: 'cors', credentials: 'same-origin' }).get().then((transactions) => {
            dispatch({
                type: SET_TRANSACTIONS,
                transactions
            });
        });
    };
};