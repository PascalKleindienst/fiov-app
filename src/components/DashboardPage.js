// ==============================================
// DASHBOARD PAGE COMPONENT =====================
// ==============================================
import React from 'react';
import TransactionList from './TransactionList';
import TransactionListFilters from './TransactionListFilters';

const DashboardPage = () => (
    <div>
        <TransactionListFilters />
        <TransactionList />
    </div>
);

export default DashboardPage;
