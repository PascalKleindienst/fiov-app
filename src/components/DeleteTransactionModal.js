// ==============================================
// DELETE TRANSACTION MODAL COMPONENT ===========
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { localize } from 'react-localize-redux';

export const DeleteTransactionModal = (props) => (
    <Modal
        isOpen={!!props.isOpen}
        contentLabel={ props.translate('transactions.modal.delete') }
        onRequestClose={props.handleClose}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="modal"
        style={{ overlay: {backgroundColor: 'rgba(0,0,0,.75)' }}}
    >
        <h3 className="modal__title">{ props.translate('transactions.modal.delete') }</h3>
        <p className="modal__body">{ props.translate('transactions.modal.confirm') }</p>
        <div className="modal__actions">
            <button className="button" onClick={props.handleConfirmation}>{ props.translate('transactions.modal.delete') }</button>
            <button className="button button--secondary" onClick={props.handleClose}>{ props.translate('cancel') }</button>
        </div>
    </Modal>
);

// PropTypes
DeleteTransactionModal.propTypes = {
    translate: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    handleConfirmation: PropTypes.func.isRequired
};

export default localize(DeleteTransactionModal, 'locale');