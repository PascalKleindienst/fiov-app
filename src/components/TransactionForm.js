// ==============================================
// TRANSACTION FORM COMPONENT ===================
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import { localize } from 'react-localize-redux';

export class TransactionForm extends React.Component {
    /**
     * Init State with props
     * @param {object} props 
     */
    constructor(props) {
        super(props);

        this.state =  {
            description: props.transaction ? props.transaction.description : '',
            note: props.transaction ? props.transaction.note : '',
            amount: props.transaction ? (props.transaction.amount / 100).toString() : '',
            createdAt: props.transaction ? moment(props.transaction.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }

    /**
     * Set description state
     */
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };

    /**
     * Set note state
     */
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };

    /**
     * Set amount state
     */
    onAmountChange = (e) => {
        const amount = e.target.value;
        
        if (!amount || amount.match(/^-?\d{0,}([\.|,]\d{0,2})?$/g)) {
            this.setState(() => ({ amount }));
        }
    };

    /**
     * Set Date state
     */
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState({ createdAt });
        }
    }
    
    /**
     * Set calendar focus state
     */
    onFocusChange = ({ focused }) => {
        this.setState({ calendarFocused: focused });
    };

    /**
     * Submit the form
     */
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount) {
            this.setState({ error: this.props.translate('transactions.form.error') });
        } else {
            // Clear the error
            this.setState({ error: '' });
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount.replace(/,/, '.'), 10) * 100,
                createdAt: this.state.createdAt.format(),
                note: this.state.note,
            });
        }
    }

    /**
     * Render Component
     */
    render() {
        return (
            <form className="form" action="#" onSubmit={this.onSubmit}>
                { this.state.error && <p className="form__error">{this.state.error}</p> }
                <input
                    type="text"
                    className="form-field"
                    placeholder={this.props.translate('transactions.form.description')}
                    autoFocus
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                />
                <input
                    type="text"
                    className="form-field"
                    placeholder={this.props.translate('transactions.form.amount')}
                    value={this.state.amount}
                    onChange={this.onAmountChange}
                />
                <SingleDatePicker
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    displayFormat="Do MMMM YYYY"
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
                <textarea
                    className="form-field form-field--textarea"
                    placeholder={this.props.translate('transactions.form.note')}
                    cols="30"
                    rows="10"
                    value={this.state.note}
                    onChange={this.onNoteChange}
                >
                </textarea>
                <div>
                    <button className="button">{ this.props.translate('transactions.form.save') }</button>
                </div>
            </form>
        )
    }
}

// PropTypes
TransactionForm.propTypes = {
    translate: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    transaction: PropTypes.object
};

export default localize(TransactionForm, 'locale');
