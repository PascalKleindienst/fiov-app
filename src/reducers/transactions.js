// ==============================================
// TRANSACTIONS REDUCERS ========================
// ==============================================
import { SET_TRANSACTIONS } from '../actions/transactions';

const transactionsReducerDefaultState = [];

export default (state = transactionsReducerDefaultState, action) => {
    switch (action.type) {
    case SET_TRANSACTIONS:
        return action.transactions;
    default:
        return state;
    }
};
