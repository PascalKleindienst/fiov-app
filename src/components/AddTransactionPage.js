// ==============================================
// ADD TRANSACTION PAGE COMPONENT ===============
// ==============================================
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getTranslate } from 'react-localize-redux';
import TransactionForm from './TransactionForm';
import { addTransaction } from '../actions/transactions';

export class AddTransactionPage extends React.Component {
    onSubmit = (transaction) => {
        this.props.addTransaction(transaction);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h2 className="page-header__title">{ this.props.translate('transactions.add_transaction') }</h2>
                    </div>
                </div>
                <div className="content-container">
                    <TransactionForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    }
}

// PropTypes
AddTransactionPage.propTypes = {
    translate: PropTypes.func.isRequired,
    addTransaction: PropTypes.func.isRequired
};

// States
const mapStateToProps = (state) => ({
    translate: getTranslate(state.locale)
});

// Dispatch
const mapDispatchToProps = (dispatch) => ({
    addTransaction: (transaction) => dispatch(addTransaction(transaction))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddTransactionPage);