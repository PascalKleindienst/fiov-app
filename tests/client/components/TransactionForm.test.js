// ==============================================
// TRANSACTION FORM TEST ========================
// ==============================================
import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { TransactionForm } from '../../../src/components/TransactionForm';
import transactions from '../fixtures/transactions';


describe('my TransactionForm component', () => {
    transactions[0].createdAt = transactions[0].createdAt.replace(/ /g, '-'); // fix for depreacted YYYY MM DD format
    let wrapperContainer;
    
    beforeEach(() => {
        wrapperContainer = shallow(<TransactionForm translate={ (msg) => msg } onSubmit={ () => false } />);
    });

    test('should render TransactionForm correctly', () => {
        expect(wrapperContainer).toMatchSnapshot();
    });

    test('should render TransactionForm with transaction data', () => {
        const wrapper = shallow(<TransactionForm transaction={transactions[0]} translate={ (msg) => msg } onSubmit={ () => false } />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render error for invalid form submission', () => {
        expect(wrapperContainer).toMatchSnapshot();

        wrapperContainer.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        expect(wrapperContainer.state('error').length).toBeGreaterThan(0);
        expect(wrapperContainer).toMatchSnapshot();
    });

    test('should set description on input change', () => {
        const value = 'New description';
        wrapperContainer.find('input').at(0).simulate('change', {
            target: { value }
        });
        expect(wrapperContainer.state('description')).toBe(value);
    });

    test('should set note on textarea change', () => {
        const value = 'New note';
        wrapperContainer.find('textarea').at(0).simulate('change', {
            target: { value }
        });
        expect(wrapperContainer.state('note')).toBe(value);
    });

    test('should set amount if valid input', () => {
        const value = '23.50';
        wrapperContainer.find('input').at(1).simulate('change', {
            target: { value }
        });
        expect(wrapperContainer.state('amount')).toBe(value);
    });

    test('should set negative amount if valid input', () => {
        const value = '-23.50';
        wrapperContainer.find('input').at(1).simulate('change', {
            target: { value }
        });
        expect(wrapperContainer.state('amount')).toBe(value);
    });

    test('should not set amount if invalid input', () => {
        const value = '12.122';
        wrapperContainer.find('input').at(1).simulate('change', {
            target: { value }
        });
        expect(wrapperContainer.state('amount')).toBe('');
    });

    test('should call onSubmit prop for valid form submission', () => {
        const onSubmitSpy = jest.fn();
        const wrapper = shallow(<TransactionForm transaction={transactions[0]} onSubmit={onSubmitSpy} translate={ (msg) => msg } />);
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        expect(wrapper.state('error')).toBe('');
        expect(onSubmitSpy).toHaveBeenLastCalledWith({
            description: transactions[0].description,
            amount: transactions[0].amount,
            createdAt: moment(transactions[0].createdAt).format(),
            note: transactions[0].note,
        });
    });

    test('should set new date on date change', () => {
        const now = moment();

        wrapperContainer.find('withStyles(SingleDatePicker)').prop('onDateChange')(now);
        expect(wrapperContainer.state('createdAt')).toEqual(now);
    });

    test('should not set a new date on date change if date is empty/false', () => {
        const wrapper = shallow(<TransactionForm transaction={transactions[0]} translate={ (msg) => msg } onSubmit={ () => false } />);

        wrapper.find('withStyles(SingleDatePicker)').prop('onDateChange')();
        expect(wrapper.state('createdAt')).toEqual(moment(transactions[0].createdAt));
    });

    test('should set calendar focus on change', () => {
        const focused = true;

        wrapperContainer.find('withStyles(SingleDatePicker)').prop('onFocusChange')({ focused });
        expect(wrapperContainer.state('calendarFocused')).toBe(focused);
    });

    test('should set isOutsideRange always to false', () => {
        const isOutsideRange = wrapperContainer.find('withStyles(SingleDatePicker)').prop('isOutsideRange');
        expect(isOutsideRange()).toBe(false);
    });
});