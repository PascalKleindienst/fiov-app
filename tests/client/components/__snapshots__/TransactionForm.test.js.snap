// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my TransactionForm component should render TransactionForm correctly 1`] = `
<form
  action="#"
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.description"
    type="text"
    value=""
  />
  <input
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat="Do MMMM YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="form-field form-field--textarea"
    cols="30"
    onChange={[Function]}
    placeholder="transactions.form.note"
    rows="10"
    value=""
  />
  <div>
    <button
      className="button"
    >
      transactions.form.save
    </button>
  </div>
</form>
`;

exports[`my TransactionForm component should render TransactionForm with transaction data 1`] = `
<form
  action="#"
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.description"
    type="text"
    value="Gum"
  />
  <input
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.amount"
    type="text"
    value="-1.99"
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:01.970Z"}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat="Do MMMM YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="form-field form-field--textarea"
    cols="30"
    onChange={[Function]}
    placeholder="transactions.form.note"
    rows="10"
    value=""
  />
  <div>
    <button
      className="button"
    >
      transactions.form.save
    </button>
  </div>
</form>
`;

exports[`my TransactionForm component should render error for invalid form submission 1`] = `
<form
  action="#"
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.description"
    type="text"
    value=""
  />
  <input
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat="Do MMMM YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="form-field form-field--textarea"
    cols="30"
    onChange={[Function]}
    placeholder="transactions.form.note"
    rows="10"
    value=""
  />
  <div>
    <button
      className="button"
    >
      transactions.form.save
    </button>
  </div>
</form>
`;

exports[`my TransactionForm component should render error for invalid form submission 2`] = `
<form
  action="#"
  className="form"
  onSubmit={[Function]}
>
  <p
    className="form__error"
  >
    transactions.form.error
  </p>
  <input
    autoFocus={true}
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.description"
    type="text"
    value=""
  />
  <input
    className="form-field"
    onChange={[Function]}
    placeholder="transactions.form.amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat="Do MMMM YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    className="form-field form-field--textarea"
    cols="30"
    onChange={[Function]}
    placeholder="transactions.form.note"
    rows="10"
    value=""
  />
  <div>
    <button
      className="button"
    >
      transactions.form.save
    </button>
  </div>
</form>
`;
